icu4c/source/i18n/calendar.cpp

```diff
@@ -856,8 +856,8 @@ Calendar::operator=(const Calendar &right)
         fWeekendCease            = right.fWeekendCease;
         fWeekendCeaseMillis      = right.fWeekendCeaseMillis;
         fNextStamp               = right.fNextStamp;
-        uprv_strncpy(validLocale, right.validLocale, sizeof(validLocale));
-        uprv_strncpy(actualLocale, right.actualLocale, sizeof(actualLocale));
+        uprv_memcpy(validLocale, right.validLocale, sizeof(validLocale));
+        uprv_memcpy(actualLocale, right.actualLocale, sizeof(actualLocale));
```

Comment:
Does it not complain about reading bytes from right.valid/actualLocale that may be uninitialized? I assume that the constructor does not usually touch every last byte there.

Reply:
https://stackoverflow.com/questions/1453876/why-does-strncpy-not-null-terminate suggest set the dec to empty string then use strncat instead

Reply:
does not work. try to use -Wno-stringop-truncation
 instead